extensions/wikia/ThemeDesigner/ThemeDesignerController.class.php

```diff
@@ -290,15 +315,13 @@ public function executeBackgroundImageUpload() {
 	 * @return array
 	 */
 	private function uploadImage($upload) {
-		global $wgRequest, $wgUser, $wgEnableUploads;
-
 		$uploadStatus = array("status" => "error");
 
-		if ( empty( $wgEnableUploads )) {
+		if ( empty( $this->wg->EnableUploads ) ) {
 			$uploadStatus["errors"] = [ wfMessage( 'themedesigner-upload-disabled' )->plain() ];
 		} else {
-			$upload->initializeFromRequest( $wgRequest );
-			$permErrors = $upload->verifyPermissions( $wgUser );
+			$upload->initializeFromRequest( $this->wg->Request );
+			$permErrors = $upload->verifyPermissions( $this->wg->User );
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Refactor this code to rely on RequestContext instead of context-dependent globals. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=wikia-php-rules%3AContextualWgVariableUsage)